openstack/dws/cluster/requests.go

```diff
@@ -6,48 +6,75 @@ import (
 	"github.com/chnsz/golangsdk"
 )
 
+const (
+	PublicBindTypeAuto         = "auto_assign"
+	PublicBindTypeNotUser      = "not_use"
+	PublicBindTypeBindExisting = "bind_existing"
+)
+
 type PublicIpOpts struct {
 	EipID          string `json:"eip_id,omitempty"`
 	PublicBindType string `json:"public_bind_type,omitempty"`
 }
 
 type CreateOpts struct {
-	Name             string        `json:"name" required:"true"`
-	NumberOfNode     int           `json:"number_of_node" required:"true"`
-	AvailabilityZone string        `json:"availability_zone,omitempty"`
-	SubnetID         string        `json:"subnet_id" required:"true"`
-	UserPwd          string        `json:"user_pwd" required:"true"`
-	SecurityGroupID  string        `json:"security_group_id" required:"true"`
-	PublicIp         *PublicIpOpts `json:"public_ip,omitempty"`
-	NodeType         string        `json:"node_type" required:"true"`
-	VpcID            string        `json:"vpc_id" required:"true"`
-	UserName         string        `json:"user_name" required:"true"`
-	Port             int           `json:"port,omitempty"`
+	Name                string        `json:"name" required:"true"`
+	NumberOfNode        int           `json:"number_of_node" required:"true"`
+	AvailabilityZone    string        `json:"availability_zone,omitempty"`
+	SubnetID            string        `json:"subnet_id" required:"true"`
+	UserPwd             string        `json:"user_pwd" required:"true"`
+	SecurityGroupID     string        `json:"security_group_id" required:"true"`
+	PublicIp            *PublicIpOpts `json:"public_ip,omitempty"`
+	NodeType            string        `json:"node_type" required:"true"`
+	VpcID               string        `json:"vpc_id" required:"true"`
+	UserName            string        `json:"user_name" required:"true"`
+	Port                int           `json:"port,omitempty"`         //default：8000
+	NumberOfCn          int           `json:"number_of_cn,omitempty"` //default：2
+	EnterpriseProjectId string        `json:"enterprise_project_id,omitempty"`
+}
+
+type ListOpts struct {
```

Comment:
according to the API doc, the List does not support any querying parameters.

Reply:
In fact, they support query by enterprise_project_id  and have confirmed with service owner

Reply:
In fact, they support query by enterprise_project_id and have confirmed with service owner